pkg/services/rpcsrv/subscription_test.go

```diff
@@ -575,39 +576,53 @@ func doSomeWSRequest(t *testing.T, ws *websocket.Conn) {
 
 func TestWSClientsLimit(t *testing.T) {
 	for tname, limit := range map[string]int{"default": 0, "8": 8, "disabled": -1} {
-		effectiveClients := limit
-		if limit == 0 {
-			effectiveClients = defaultMaxWebSocketClients
-		} else if limit < 0 {
-			effectiveClients = 0
-		}
 		t.Run(tname, func(t *testing.T) {
+			effectiveClients := limit
+			if limit == 0 {
+				effectiveClients = defaultMaxWebSocketClients
+			} else if limit < 0 {
+				effectiveClients = 0
+			}
 			_, _, httpSrv := initClearServerWithCustomConfig(t, func(cfg *config.Config) {
 				cfg.ApplicationConfiguration.RPC.MaxWebSocketClients = limit
 			})
 
 			dialer := websocket.Dialer{HandshakeTimeout: 5 * time.Second}
 			url := "ws" + strings.TrimPrefix(httpSrv.URL, "http") + "/ws"
-			wss := make([]*websocket.Conn, effectiveClients)
-
-			for i := 0; i < len(wss)+1; i++ {
-				ws, r, err := dialer.Dial(url, nil)
-				if r != nil && r.Body != nil {
-					defer r.Body.Close()
-				}
-				if i < effectiveClients {
+			wss := make([]*websocket.Conn, 0, effectiveClients)
+			var mu sync.Mutex // To safely append to wss slice
+			var wg sync.WaitGroup
+
+			// Dial effectiveClients connections in parallel
+			for i := 0; i < effectiveClients; i++ {
+				wg.Add(1)
+				go func() {
+					defer wg.Done()
+					ws, r, err := dialer.Dial(url, nil)
+					if r != nil {
+						defer r.Body.Close()
+					}
 					require.NoError(t, err)
-					wss[i] = ws
-					// Check that it's completely ready.
-					doSomeWSRequest(t, ws)
-				} else {
-					require.Error(t, err)
-				}
+					mu.Lock()
+					wss = append(wss, ws)
+					mu.Unlock()
+				}()
 			}
+
+			wg.Wait() // Wait for all successful connections
+
+			// Attempt one more connection, which should fail
+			_, r, err := dialer.Dial(url, nil)
+			require.Error(t, err, "The connection beyond the limit should fail")
+			if r != nil {
+				r.Body.Close()
+			}
```

Comment:
It's always nil for failing connection, isn't it? Check it.

Reply:
no, response is:
`&http.Response{Status:"500 Internal Server Error", StatusCode:500, Proto:"HTTP/1.1", ProtoMajor:1, ProtoMinor:1, Header:http.Header{"Content-Length":[]string{"118"}, "Content-Type":[]string{"application/json; charset=utf-8"}, "Date":[]string{"Tue, 02 Apr 2024 12:04:43 GMT"}}, Body:io.nopCloserWriterTo{Reader:(*bytes.Reader)(0x140002bfa70)}, ContentLength:118, TransferEncoding:[]string(nil), Close:false, Uncompressed:false, Trailer:http.Header(nil), Request:(*http.Request)(0x140007a0200), TLS:(*tls.ConnectionState)(nil)}`

Reply:
OK.